hazelcast-jet-core/src/test/java/com/hazelcast/jet/job/JobSubmissionSlownessRegressionTest.java

```diff
@@ -0,0 +1,162 @@
+    private static final int WAIT_BEFORE_MEASUREMENT_B_SECTION
+            = MEASUREMENT_A_CYCLE_SECTION + WAIT_BEFORE_MEASUREMENT_B_COUNT;
+    private static final int MEASUREMENT_B_CYCLE_SECTION = WAIT_BEFORE_MEASUREMENT_B_SECTION + MEASUREMENT_B_CYCLE_COUNT;
+
+    private static int measurementCount;
+
+    @Before
+    public void setup() {
+        JetConfig config = new JetConfig();
+        config.getHazelcastConfig().setProperty("hazelcast.logging.type", "none");
+        createJetMember(config);
+    }
+
+    @Test
+    public void regressionTestForPR1488() throws ExecutionException, InterruptedException {
+        ExecutorService executorService = Executors.newFixedThreadPool(THREADS_COUNT);
+
+        double measurementARateSum = 0;
+        double measurementBRateSum = 0;
+
+        DAG dag = twoVertex();
+        JetInstance client = createJetClient();
+        while (measurementCount <= MEASUREMENT_B_CYCLE_SECTION) {
+            AtomicInteger completedRoundtrips = new AtomicInteger();
+            long start = System.nanoTime();
+            List<Future> futures = new ArrayList<Future>();
+            for (int i = 0; i < THREADS_COUNT; i++) {
+                Future<?> f = executorService.submit(() -> {
+                    bench(DURATION_SECS, () -> {
+                        client.newJob(dag, new JobConfig()).join();
+                    }, completedRoundtrips);
+                });
+                futures.add(f);
+            }
+            for (Future future : futures) {
+                future.get();
+            }
+            long elapsed = System.nanoTime() - start;
+            double rate = (double) completedRoundtrips.get() / (double) elapsed * TimeUnit.SECONDS.toNanos(1);
+            System.out.println("Rate was " + rate + " req/s");
+            measurementCount++;
+            if (measurementCount <= HEAT_UP_CYCLE_SECTION) {
+            } else {
+                if (measurementCount <= MEASUREMENT_A_CYCLE_SECTION) {
+                    measurementARateSum += rate;
+                } else {
+                    if (measurementCount <= WAIT_BEFORE_MEASUREMENT_B_SECTION) {
+                    } else {
+                        if (measurementCount <= MEASUREMENT_B_CYCLE_SECTION) {
+                            measurementBRateSum += rate;
+                        }
+                    }
+                }
+            }
+        }
+
+        double measurementARate = measurementARateSum / MEASUREMENT_A_CYCLE_COUNT;
+        double measurementBRate = measurementBRateSum / MEASUREMENT_B_CYCLE_COUNT;
+
+        assertTrue("Job submission rate should not decrease. First rate: " + measurementARate
+                + ", second rate: " + measurementBRate, measurementARate * 0.8 < measurementBRate);
+    }
+
+    private static DAG twoVertex() {
+        DAG dag = new DAG();
+        Vertex v1 = dag.newVertex("v", () -> new SleepingP(0));
```

Comment:
we can remove `SleepingP` and replace it with `Processors.noop()`. It was originally used to test when the processor was not completing immediately.

Reply:
You're right. I updated it.